### PR TITLE
Add support for PayPal recurring and 3DS

### PR DESCRIPTION
**Pull request by EpokK**
- Update PayPal view with the new payment token logic
- Update payment methods table
- Update payment methods select

### CODE REVIEW

**Comment by emnproton** [comment]
The gateway page at secure.protonmail.blue/com does not auto-close once the transaction is successfully completed.

Unless the user closes the tab, they have no way of knowing that their payment went through.

**Video:**

[secure page doesnt auto close.zip](https://github.com/ProtonMail/react-components/files/3477360/secure.page.doesnt.auto.close.zip)

File: containers/payments/PayPal.js [hunk 1]
```diff
@@ -1,43 +1,176 @@
...
+    const [loading, withLoading] = useLoading();
+    const tabRef = useRef(null);
+    const timerRef = useRef(0);
+    const processRef = useRef(true);
+    const tokenRef = useRef('');
+    const api = useApi();
+    const [approvalURL, setAppovalURL] = useState('');
+    const [error, setError] = useState(null);
+
+    const I18N = {
```
Comment (Yiin) [inline containers/payments/PayPal.js]: It would be better to move this outside the component, to avoid repeated calls on re-renders.

Comment (EpokK) [reply containers/payments/PayPal.js]: This needs to be inside the component to refresh it when the language is changed.

File: containers/payments/tokenHelper.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+        const onMessage = (event) => {
+            const origin = event.origin || event.originalEvent.origin; // For Chrome, the origin property is in the event.originalEvent object.
+
+            if (origin !== 'https://secure.protonmail.com') {
+                return reject();
+            }
+
+            if (event.source !== tab) {
+                return reject();
```
Comment (mmso) [inline containers/payments/tokenHelper.js]: I don't think we need to `reject()` here or at the origin check

Comment (EpokK) [reply containers/payments/tokenHelper.js]: But if I don't `reject` here, who take care of the promise process? I don't like to count on `listenTab` for it.

Comment (EpokK) [reply containers/payments/tokenHelper.js]: Oh I understand your point now. You are right.

**Pull request merged**